### PR TITLE
Emailing

### PR DESCRIPTION
**Pull request by TheMightyRaider**

### CODE REVIEW

File: controllers/email.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      to: 'sivasanjay7@gmail.com',
+      subject: 'Registration Success',
+      // text: `Welcome ${user.first_name}`,
+      html: '<h1>Explore the World! <h1> <br> ',
+      // template: 'index',
+    }
+
+    // sendMail to send the mail
+
+    transporter.sendMail(mailOptions, function(error, info) {
```
Comment (digi0ps) [inline controllers/email.js]: Not immediately, but try to abstract this function and return a promise.
Like you should be able to write the below code across all functions.
```
send(mailOptions).then()
```

**Comment by digi0ps** [comment]
Here is the pug template for Welcome email:
```
doctype html
html
  head
    <link href="https://fonts.googleapis.com/css?family=Montserrat" rel="stylesheet" type="text/css">
    style.
      * {
        font-family: 'Montserrat', sans-serif;
        color: #2a2a2a;
      }

      body {
        background-color: #fafafa;
      }

  body
    .container
      center
        img(
          width="500px"
          src="https://www.sanjay7.com/assets/eiffel-tower.svg"
        )
        h1 Welcome to Visiit, #{name}.
        p Let's start exploring together
```

File: controllers/email.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    //     viewPath: '../views/',
+    //     extName: '.hbs',
+    //   }),
+    // )
+
+    // mailOptions contains the message to be sent to the registered user
+
+    var mailOptions = {
+      from: 'isha.venkatesh06@gmail.com',
+      to: 'sivasanjay7@gmail.com',
```
[the remark this answers is not in the document]
Comment (TheMightyRaider) [reply controllers/email.js]: Yeah ,`to` has be declared for checking purpose.

[the remark this answers is not in the document]
Comment (digi0ps) [reply controllers/email.js]: `to`: `user.email`

File: controllers/email.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const nodemailer = require('nodemailer')
+const email = require('email-templates')
```
Comment (digi0ps) [inline controllers/email.js]: Package variables must start with a capital letter.

File: controllers/email.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  service: 'gmail',
+  auth: {
+    user: 'haventcreatedyet@gmail.com',
+    pass: 'password',
+  },
+})
+
+module.exports = {
+  sendWelcomeMail: function(user) {
+    const Email = new email({
```
Comment (digi0ps) [inline controllers/email.js]: Variable names must start with small letter.

File: controllers/User.js [hunk 5]
```diff
@@ -18,8 +19,11 @@ exports.postUser = (req, res) => {
       else user.password = hash
       user
         .save()
-        .then(() => {
-          res.send(success(filterUser(user.toObject())))
+        .then(user => {
+          req.session.userId = user._id
+          // res.send(success(filterUser(user.toObject())))
+          res.redirect('/api/v1')
```
Comment (digi0ps) [inline controllers/User.js]: no need for redirect FE will take care

Comment (TheMightyRaider) [reply controllers/User.js]: FE  ? @digi0ps

**Pull request merged**